tests/unit/generators/xml_files/test_create_configuration_xml.py

```diff
@@ -1,22 +1,10 @@
-from unittest.mock import patch, MagicMock
 from splunk_add_on_ucc_framework.generators.xml_files import ConfigurationXml
+import os.path
+import xmldiff.main
 
+from splunk_add_on_ucc_framework import __file__ as ucc_framework_file
 
-@patch(
-    "splunk_add_on_ucc_framework.data_ui_generator.generate_views_configuration_xml",
-    return_value="<xml></xml>",
-)
-def test_set_attributes(
-    mock_generate_xml, global_config_all_json, input_dir, output_dir, ucc_dir, ta_name
-):
-    config_xml = ConfigurationXml(
-        global_config_all_json,
-        input_dir,
-        output_dir,
-        ucc_dir=ucc_dir,
-        addon_name=ta_name,
-    )
-    assert hasattr(config_xml, "configuration_xml_content")
+UCC_DIR = os.path.dirname(ucc_framework_file)
 
 
 def test_set_attributes_without_configuration(
```

Comment:
I think this can be even further simplified, `_set_attributes` can be "removed" from the `ConfigurationXml` implementation and then we would have only 2 tests (when there is no configuration and where there is configuration).

Reply:
Same for all files below.

Reply:
I think we cannot. This would then conflict with your vision for AppConf and AppManifest classes.

Reply:
can you please elaborate more?

`AppManifest` class seems to be untouched for some time and `AppConf` can benefit from the same improvements I mentioned above.

Reply:
In our previous discussions related to merging of `app.manifest` and `app.conf` with GlobalConfig, we had discussed passing some objects, provide it to the class and the class object then has all the attributes required of the given class. 

Further, we can do everything in "__init__" directly, but the purpose or logic of this new design was to define what should be done where, hence maintaining the modularization of the code.